packages/jsActions/mobile-resources-native/src/file-download/DownloadFile.ts

```diff
@@ -60,22 +76,19 @@ export async function DownloadFile(file: mendix.lib.MxObject, openWithOS: boolea
         accessiblePath = await getUniqueFilePath(baseDir, sanitizedFileName);
         await RNBlobUtil.fs.cp(filePath, accessiblePath);
     } else {
-        const tempPath = await getUniqueFilePath(baseDir, sanitizedFileName);
-        const base64Data = await mx.readFileBlob(filePath);
-        const base64Content = base64Data?.split(",")[1];
-        await RNBlobUtil.fs.createFile(tempPath, base64Content, "base64");
-        const mimeType = mimeTypes.getType(fileName) as Mediatype;
-        accessiblePath = await RNBlobUtil.MediaCollection.copyToMediaStore(
-            {
-                name: sanitizedFileName,
-                mimeType: mimeType ?? "*",
-                parentFolder: ""
-            },
-            "Download",
-            tempPath
-        );
-
-        RNBlobUtil.fs.unlink(tempPath);
+        const mimeType = mimeTypes.getType(fileName);
+        const copyToMediaStore = createCopyToMediaStoreFunction(sanitizedFileName, mimeType);
+        if (typeof mx.readFileBlob === "function") {
+            const tempPath = await getUniqueFilePath(baseDir, sanitizedFileName);
+            const base64Data = await mx.readFileBlob(filePath);
+            const base64Content = base64Data?.split(",")[1];
+            await RNBlobUtil.fs.createFile(tempPath, base64Content, "base64");
+            accessiblePath = await copyToMediaStore(tempPath);
```

Comment:
Why to store the file into tempPath? The other `copyToMediaStore` (line 90) still stores at filePath.